public/docs/_examples/architecture/dart/lib/sales_tax_component.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:angular2/core.dart';
+
+import 'sales_tax_service.dart';
+import 'tax_rate_service.dart';
+
+@Component(
+    selector: 'sales-tax',
+    template: '''
+      <h2>Sales Tax Calculator</h2>
+      Amount: <input #amountBox (input)="0" type="number">
```

Comment:
Triggering on `onInput` rather than `onChange` makes for a more responsive user experience (UX). Again, using `type="number"` makes for a better UX in those browsers that support it, and makes no difference (from the app processing point of view) in those browsers that do not support it.